subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -47,6 +54,25 @@ static bool set_second(struct tm *sched_time,
 		       struct bt_mesh_schedule_entry *entry,
 		       struct bt_mesh_time_srv *srv);
 
+static int store(struct bt_mesh_scheduler_srv *srv, uint8_t idx)
+{
+	char name[2] = {0};
+
+	if (!IS_ENABLED(CONFIG_BT_SETTINGS)) {
```

Comment:
Could you conditionally call `store` instead of checking feature enabling here?
My assumption is if you call `store` conditionally the linker will cut out it when `CONFIG_BT_SETTINGS` is off.

Reply:
In principle, I agree. However, this is the pattern used in many models, why do it differently here?

Reply:
If you feel this solution is better leave it as is. I think any template is not dogma and can be changed at any time if a suggested solution is more optimized.

Reply:
If I remember correctly, this manual optimization is not necessary. The compiler will end up producing an empty function here when it realizes that it'll unconditionally exit early. This gets inlined and noop'd. It'd be different if the function wasn't static though.